metadata-ingestion/src/datahub/ingestion/source/sample_source.py

```diff
@@ -0,0 +1,25 @@
+import pathlib
+
+from datahub.cli.docker_cli import download_sample_data
+from datahub.configuration.common import ConfigModel
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.api.decorators import (
+    SupportStatus,
+    config_class,
+    platform_name,
+    support_status,
+)
+from datahub.ingestion.source.file import FileSourceConfig, GenericFileSource
+
+
+@platform_name("SampleData")
+@config_class(ConfigModel)
+@support_status(SupportStatus.UNKNOWN)
+class SampleDataSource(GenericFileSource):
```

Comment:
Can we generalize this to be an "HttpFileSource" which takes a URL location, downloads the file, and then ingests from that. This might be useful later on when loading data from one gms into another...? Not sure

Reply:
Suggestion from Tamas: use smart-open https://pypi.org/project/smart-open/ so that we can easily support a variety of uri types e.g. s3, gcp, http

Reply:
hey folks
Great suggestions. But the file source does a lot more than just opening a file. It supports ingesting from folders too, reporting size of each file in folder, number of files etc. 
I tried adding `smart-open` to see what will happen and it adds a lot more backtracking to the install process. Unless we start freezing dependencies this will require more work.

I am not switching to a HttpFileSource here. We can take that elsewhere.